docs/requirements.txt

```diff
@@ -7,3 +7,4 @@ sphinx-copybutton==0.5.2
 sphinx-autobuild==2021.3.14
 myst-parser
 sphinx-pyproject
+pre-commit
```

Comment:
You don't have to add pre-commit requirement here.
It's included in pyproject.toml's optional-dependencies.

Pls check how to install optional-dependencies(dev) [here](https://optinist.readthedocs.io/en/latest/installation/each_platforms_for_developer.html#install-requirements)